File: 002_euclidean_algorithm/implementations/python/main.py
def gcd(a, b):
    while b > 0:
        a %= b
        a, b = b, a
    return a


def gcd_recurs(a, b):
    if b == 0:
        return a
    return gcd_recurs(b, a % b)


if __name__ == "__main__":
    print(gcd(120, 140))
    print(gcd_recurs(120, 140))

File: 002_euclidean_algorithm/implementations/c/main.c
#include <stdio.h>

int gcd(int a, int b)
{
    while (b > 0)
    {
        a %= b;
        // swap both the numbers
        a += b;
        b = a - b;
        a = a - b;
    }
    return a;
}

int gcd_recurs(int a, int b)
{
    if (b == 0)
        return a;
    return gcd_recurs(b, a % b);
}

int main()
{
    int a = 120;
    int b = 140;
    printf("%d\n", gcd(a, b));
    printf("%d\n", gcd_recurs(a, b));

    return 0;
}